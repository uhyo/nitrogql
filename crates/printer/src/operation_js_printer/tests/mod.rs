#![cfg(test)]

use insta::assert_snapshot;
use nitrogql_ast::OperationDocument;
use nitrogql_parser::parse_operation_document;
use nitrogql_semantics::resolve_operation_extensions;
use sourcemap_writer::JustWriter;

use crate::{print_js_for_operation_document, OperationJSPrinterOptions};

#[test]
fn print_query() {
    let document = parse(
        r#"
        query MyQuery {
            user {
                id
                name
            }
        }
    "#,
    );

    assert_snapshot!(print_js(&document));
}

#[test]
fn print_fragment_only() {
    let document = parse(
        r#"
        fragment Foo on User {
            id name
        }
    "#,
    );

    assert_snapshot!(print_js(&document));
}

#[test]
fn print_fragment_and_query() {
    let document = parse(
        r#"
        query MyQuery {
            user {
                ...Foo
            }
        }

        fragment Foo on User {
            id name
        }
    "#,
    );

    assert_snapshot!(print_js(&document))
}

#[test]
fn print_query_with_variables() {
    let document = parse(
        r#"
        query MyQuery($id: ID!) {
            user(id: $id) {
                id
                name
            }
        }
    "#,
    );

    assert_snapshot!(print_js(&document));
}

#[test]
fn print_nested_fragments() {
    let document = parse(
        r#"
        query MyQuery {
            user {
                ...Foo
            }
        }

        fragment Foo on User {
            id
            ...Bar
        }

        fragment Bar on User {
            name
        }
    "#,
    );

    assert_snapshot!(print_js(&document));
}

fn parse(str: &str) -> OperationDocument {
    let doc = parse_operation_document(str).unwrap();
    let (document, _) = resolve_operation_extensions(doc).unwrap();
    document
}

fn print_js(document: &OperationDocument) -> String {
    let mut buffer = String::new();
    let mut writer = JustWriter::new(&mut buffer);
    let options = OperationJSPrinterOptions::default();
    print_js_for_operation_document(options, document, &mut writer);
    buffer
}
