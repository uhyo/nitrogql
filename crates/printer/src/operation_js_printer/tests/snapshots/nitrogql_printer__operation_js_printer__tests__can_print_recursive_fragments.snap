---
source: crates/printer/src/operation_js_printer/tests/mod.rs
expression: print_js(&document)
---
const MyQueryQuery = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"query","name":{"kind":"Name","value":"MyQuery"},"variableDefinitions":[],"directives":[],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"user"},"arguments":[],"directives":[],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"FragmentSpread","name":{"kind":"Name","value":"Foo"},"directives":[]}]}}]}},{"kind":"FragmentDefinition","name":{"kind":"Name","value":"Foo"},"typeCondition":{"kind":"NamedType","name":{"kind":"Name","value":"User"}},"directives":[],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"},"arguments":[],"directives":[]},{"kind":"Field","name":{"kind":"Name","value":"user"},"arguments":[],"directives":[],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"FragmentSpread","name":{"kind":"Name","value":"Foo"},"directives":[]}]}}]}}]};

export { MyQueryQuery as default };

export const Foo = {"kind":"Document","definitions":[{"kind":"FragmentDefinition","name":{"kind":"Name","value":"Foo"},"typeCondition":{"kind":"NamedType","name":{"kind":"Name","value":"User"}},"directives":[],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"},"arguments":[],"directives":[]},{"kind":"Field","name":{"kind":"Name","value":"user"},"arguments":[],"directives":[],"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"FragmentSpread","name":{"kind":"Name","value":"Foo"},"directives":[]}]}}]}}]};


